packages/connect/src/impl/core-in-module.ts

```diff
@@ -118,6 +121,10 @@ export class CoreInModule implements ConnectFactoryDependencies<ConnectSettingsP
 
         if (type === POPUP.CANCEL_POPUP_REQUEST) return;
 
+        if (this.onCoreHook) {
+            message = this.onCoreHook(message);
```

Comment:
This is working correctly as we're modifying the nested payload in this case, but if we reassign it instead, it could be inconsistent with the `const { event, type, payload } = message` spread above, couldn't it? Wouldn't it be safer to call the hook right after the `cloneObject`?